pkg/sinks/sink.go

```diff
@@ -129,7 +130,10 @@ func (u *SinkProcessor) Start(ctx context.Context) error {
 
 			// create watermark publisher using watermark stores
 			publishWatermark = jetstream.BuildPublishersFromStores(ctx, u.VertexInstance, sinkWmStores)
+			// sink vertex has only one toBuffer, so the length is 1
+			idleManager = wmb.NewIdleManager(1)
```

Comment:
does sink need a publisher?

Reply:
Yes because UI needs to fetch the sink watermark.